monitoring/src/check-page/common-functions.ts

```diff
@@ -46,33 +47,94 @@ export const makeNewBrowser = async (debugMode: boolean): Promise<Browser> => {
 	return browser;
 };
 
+export const openPrivacySettingsPanel = async (config: Config, page: Page) => {
+	log_info(`Loading privacy settings panel: Start`);
+	// Ensure that Sourcepoint has enough time to load the CMP
+	await page.waitForTimeout(3000);
+
+	const frame = getFrame(page, config.iframeDomain);
+	await frame.click(ELEMENT_ID.TCFV2_FIRST_LAYER_MANAGE_COOKIES);
+
+	log_info(`Loading privacy settings panel: Complete`);
+};
+
+export const checkPrivacySettingsPanelIsOpen = async (
+	config: Config,
+	page: Page,
+): Promise<void> => {
+	await page.waitForTimeout(3000);
+	log_info(`Waiting for Privacy Settings Panel: Start`);
+	const frame = getFrame(page, config.iframeDomainSecondLayer);
+	await frame.waitForSelector(ELEMENT_ID.TCFV2_SECOND_LAYER_HEADLINE);
+	log_info(`Waiting for Privacy Settings Panel: Complete`);
+};
+
+export const clickSaveAndCloseSecondLayer = async (
+	config: Config,
+	page: Page,
+) => {
+	log_info(`Clicking on save and exit button: Start`);
+	// Ensure that Sourcepoint has enough time to load the CMP
+	await page.waitForTimeout(5000);
+
+	const frame = getFrame(page, config.iframeDomainSecondLayer);
+	await frame.click(ELEMENT_ID.TCFV2_SECOND_LAYER_SAVE_AND_EXIT);
+
+	log_info(`Clicking on save and exit button: Complete`);
+};
+
+export const clickRejectAllSecondLayer = async (config: Config, page: Page) => {
+	log_info(`Clicking on reject all button: Start`);
```

Comment:
I've seen a neat way of setting log levels by overriding functions on `console` that might be handy here (there are examples in the `tracker` and `github-lens` repositories.

Reply:
This looks like it might require a refactor and this PR is now blocking a couple of others so I have created a card in the backlog to look at this later.